crates/tessera-backbone/src/database/mod.rs

```diff
@@ -0,0 +1,251 @@
+        db_hostname = db_hostname,
+        port = port,
+        db_user = db_username
+    );
+
+    let signable_request =
+        SignableRequest::new("GET", &url, std::iter::empty(), SignableBody::Bytes(&[])).expect("signable request");
+
+    let (signing_instructions, _signature) = sign(signable_request, &signing_params.into())?.into_parts();
+
+    let mut url = url::Url::parse(&url).unwrap();
+    for (name, value) in signing_instructions.params() {
+        url.query_pairs_mut().append_pair(name, value);
+    }
+
+    let response = url.to_string().split_off("https://".len());
+
+    Ok(response)
+}
+
+pub async fn connect_to_database(
+    host: &str,
+    port: u16,
+    database_name: &str,
+    auth: &AuthMethod,
+) -> anyhow::Result<Arc<DatabaseConnection>> {
+    let mut options = match auth {
+        AuthMethod::Credential { username, password } => {
+            let mut conn_str = Url::parse(&format!("postgres://{host}:5432/{database_name}?sslmode=Prefer"))?;
+            conn_str.set_username(username).unwrap();
+            conn_str.set_password(password.as_deref()).unwrap();
+            ConnectOptions::new(conn_str)
+        }
+        AuthMethod::RdsIamAuth { host: auth_host, port: auth_port, username } => {
+            let password = generate_rds_iam_token(auth_host, *auth_port, username).await?;
+            let mut conn_str = Url::parse(&format!("postgres://{host}:{port}/postgres?sslmode=Prefer"))?;
+            conn_str.set_username(username).unwrap();
+            conn_str.set_password(Some(urlencoding::encode(&password).as_ref())).unwrap();
+            ConnectOptions::new(conn_str)
+        }
+    };
+
+    options.sqlx_logging_level(tracing::log::LevelFilter::Debug);
+
+    let connection = Arc::new(Database::connect(options).await?);
+
+    if let AuthMethod::RdsIamAuth { host: auth_host, port: auth_port, username } = auth {
+        reassign_token_periodically_to_database(connection.clone(), auth_host.clone(), *auth_port, username.clone());
+    };
+
+    Ok(connection)
+}
+
+fn reassign_token_periodically_to_database(
+    database: Arc<DatabaseConnection>,
+    database_host: String,
+    database_port: u16,
+    database_user: String,
+) {
+    tokio::spawn({
+        async move {
+            loop {
+                tokio::time::sleep(tokio::time::Duration::from_secs(600)).await;
+                if let Ok(token) = generate_rds_iam_token(&database_host, database_port, &database_user).await {
+                    let pool = database.get_postgres_connection_pool();
+                    let new_option = PgConnectOptions::new()
+                        .host(&database_host)
+                        .database("postgres")
+                        .username(&database_user)
+                        .password(&token);
+                    pool.set_connect_options(new_option);
+                }
+            }
+        }
+    });
+}
+
+pub trait OrganizationScopedTransaction {
```

Comment:
The `OrganizationScopedTransaction` entity also appears to require the `Send` trait bound.